pkg/env/env.go

```diff
@@ -52,19 +52,30 @@ func NewEnv(ctx context.Context, log *logrus.Entry) (Interface, error) {
 		return nil, err
 	}
 
-	if strings.ToLower(os.Getenv("RP_MODE")) == "int" {
-		log.Warn("running in int mode")
-		return newInt(ctx, log, im, clientauthorizer.NewARM(log), clientauthorizer.NewAdmin(log))
-	}
-
 	for _, key := range []string{
 		"MDM_ACCOUNT",
 		"MDM_NAMESPACE",
+		"ADMIN_API_CA_BUNDLE_PATH",
+		"ADMIN_API_CLIENT_CERT_COMMON_NAME",
 	} {
 		if _, found := os.LookupEnv(key); !found {
 			return nil, fmt.Errorf("environment variable %q unset", key)
 		}
 	}
 
-	return newProd(ctx, log, im, clientauthorizer.NewARM(log), clientauthorizer.NewAdmin(log))
+	adminClientAuthorizer, err := clientauthorizer.NewAdmin(
```

Comment:
oh, I see, this really is forcing a tricky issue.  I think we need to separate this a bit so that when aro-monitor calls NewEnv in prod, it doesn't go through all of this code path.  There's actually more items in this category - currently aro-monitor pulls secrets which it never needs at startup because aro-rp does, and it shouldn't do that either.  Don't need to fix the whole issue in this PR, but try not to add a pattern which wouldn't work for the secrets problem as well.

Perhaps stop taking the clientauthorizers as input to the constructor and instead initializing them on demand in {Arm,Admin}ClientAuthorizer() instead?  Consider sync/Once.